srtp-kdf.go

```diff
@@ -76,10 +78,35 @@ func (kdf KDF) DeriveForStream(cipher CipherID) ([]byte, []byte, error) {
 		keySize = 32
 		saltSize = 12
 	default:
-		return nil, nil, fmt.Errorf("Unsupported cipher: %04x", cipher)
+		return 0, 0, fmt.Errorf("Unsupported cipher: %04x", cipher)
+	}
+
+	return keySize, saltSize, nil
+}
+
+func (kdf KDF) DeriveForStream(cipher CipherID) ([]byte, []byte, error) {
+	keySize, saltSize, err := kdf.getKeySize(cipher)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	key := kdf.Derive(Ke, 0, 0, keySize)
-	salt := kdf.Derive(Ks, 0, 0, saltSize)
+	// TODO replace them with actual values
```

Comment:
It turns out this TODO is wrong.  While RFC 3711 has a slot for the packet index the key derivation, in practice it is always set to zero.  So you should be able to just have `DeriveForRTP` and `DeriveForRTCP` methods, or if they're always called at the same time, one `DeriveForStream` method that derives everything.